File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>>res;
    void help(int index , vector<int>&path , vector<int>&nums){
        if(index == nums.size()){
            res.push_back(path);
            return ;
        }
        path.push_back(nums[index]);
        help(index+1,path,nums);
        path.pop_back();
        help(index+1,path,nums);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        int index = 0;
        vector<int>path;
        help(index,path,nums);
        return res;
    }
};

File: 0078-subsets/0078-subsets.java
class Solution {
    
    List<List<Integer>>res = new ArrayList<>();
    
    public void help(int index , int[] nums , ArrayList<Integer>path){
        
        if(index == nums.length){
            res.add(new ArrayList<>(path));
            return;
        }
        
        
        path.add(nums[index]);
        help(index+1,nums,path);
        path.remove(path.size() - 1);
        help(index+1,nums,path);
    }
    
    
    public List<List<Integer>> subsets(int[] nums) {
        int index = 0;
        ArrayList<Integer> path = new ArrayList<>();
        help(index,nums,path);
        return res;
        
    }
}